server/src/test/java/org/opensearch/cluster/routing/remote/NoopRemoteRoutingTableServiceTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing.remote;
+
+import org.opensearch.action.LatchedActionListener;
+import org.opensearch.cluster.Diff;
+import org.opensearch.cluster.routing.IndexRoutingTable;
+import org.opensearch.cluster.routing.RoutingTable;
+import org.opensearch.common.util.TestCapturingListener;
+import org.opensearch.gateway.remote.ClusterMetadataManifest;
+import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Before;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class NoopRemoteRoutingTableServiceTests extends OpenSearchTestCase {
+    private NoopRemoteRoutingTableService service;
+
+    @Before
+    public void setup() {
+        service = new NoopRemoteRoutingTableService();
+    }
+
+    public void testGetAsyncIndexRoutingWriteAction() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+        TestCapturingListener<ClusterMetadataManifest.UploadedMetadata> capturingListener = new TestCapturingListener<>();
```

Comment:
Instead of this you can pass a listener with `onResponse`/`onFailure` overridden to set an Atomic result. Then all you can do is check whether that result was set or not after await

Reply:
Actually, I added this Listener in #14394. Basically does the same thing by overriding the `onResponse`/`onFailure`, but it isn't atomic that shouldn't be issue for our usecase I believe.